src/test/java/com/google/research/bleth/simulator/BoardTest.java

```diff
@@ -0,0 +1,307 @@
+import static org.mockito.Mockito.when;
+
+import org.junit.runner.RunWith;
+import org.junit.Test;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.mockito.Mock;
+
+@RunWith(MockitoJUnitRunner.class)
+public final class BoardTest{
+    private static final Location negativeRowCoordinate = new Location(-1, 0);
+    private static final Location negativeColCoordinate = new Location(0, -1);
+    private static final Location zeroOnZeroCoordinate = new Location(0, 0);
+    private static final Location oneOnOneCoordinate = new Location(1, 1);
+    private static final Location zeroOnOneCoordinate = new Location(0, 1);
+    private static final Location oneOnZeroCoordinate = new Location(1, 0);
+
+    @Mock
+    private Agent firstAgent;
+    @Mock
+    private Agent secondAgent;
+
+    private boolean isBoardEmpty(Board board, int rowNum, int colNum) {
+        for (int row = 0; row < rowNum; row++) {
+            for (int col = 0; col < colNum; col++) {
+                if (!board.matrix[row][col].isEmpty()) {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    @Test
+    public void newBoardIsEmpty() {
+        Board board = new Board(2, 2);
+
+        assertThat(isBoardEmpty(board, 2, 2)).isTrue();
+    }
+
+    @Test
+    public void placeAnAgentOnBoardAgentShouldBeOnRightLocation() {
+        Board board = new Board(2, 2);
+        when(firstAgent.move()).thenReturn(zeroOnZeroCoordinate);
+
+        board.placeAgent(firstAgent.move(), firstAgent);
+
+        assertThat(board.matrix[0][0]).containsExactly(firstAgent);
+    }
+
+    @Test
+    public void placeAnAgentOnBoardAgentShouldNotBeOnOtherLocation() {
+        Board board = new Board(2, 2);
+        when(firstAgent.move()).thenReturn(oneOnOneCoordinate);
+
+        board.placeAgent(firstAgent.move(), firstAgent);
+
+        assertThat(board.matrix[0][0]).isEmpty();
+    }
+
+    @Test
+    public void placeTwoAgentsOnDifferentLocation() {
+        Board board = new Board(2, 2);
+        when(firstAgent.move()).thenReturn(zeroOnZeroCoordinate);
+        when(secondAgent.move()).thenReturn(oneOnOneCoordinate);
+
+        board.placeAgent(firstAgent.move(), firstAgent);
+        board.placeAgent(secondAgent.move(), secondAgent);
+
+        assertThat(board.matrix[0][0]).containsExactly(firstAgent);
+        assertThat(board.matrix[1][1]).containsExactly(secondAgent);
+    }
+
+    @Test
+    public void placeTwoAgentsOnSameLocation() {
+        Board board = new Board(2, 2);
+        when(firstAgent.move()).thenReturn(zeroOnOneCoordinate);
+        when(secondAgent.move()).thenReturn(zeroOnOneCoordinate);
+
+        board.placeAgent(firstAgent.move(), firstAgent);
+        board.placeAgent(secondAgent.move(), secondAgent);
+
+        assertThat(board.matrix[0][1]).containsExactly(firstAgent, secondAgent);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
```

Comment:
Prefer to use JUnit's assertThrows() instead of "expected" exceptions. See https://howtodoinjava.com/junit5/expected-exception-example/

(Note that use of "expected = FooException" in @Test is _strongly discouraged_ in Google).

Advantages of assertThrows():
* You can be sure the assertion was thrown in the specific statement, and not e.g. in some preparation step.
* You can assert further details about the thrown exception (e.g. verify the message).

Disadvantages of expected exceptions:
* If your test has any assertions after the statement that threw the exception, they won't be run at all (so the test will pass even if the assertions actually would have failed).